### DISCUSSION TITLE
PNG - How to maintain transparency

### DISCUSSION THREAD

**Discussion by yellowbooth** [ImageMagick/ImageMagick#3792] [Digital Image Processing]
Hi folks

I am trying to add brightness to PNG images that already have a fully transparent background as follows, but the resulting output has a transparent grey background.

magick  in.png -brightness-contrast 20x00  out.png

Please could you let me know how to make the resulting image also completely transparent.

Thank you :-)

**Comment by snibgo** [comment]
See the documentation https://imagemagick.org/script/command-line-options.php#brightness-contrast :
```
You may also use -channel to control which channels to apply the brightness and/or contrast change.
 The default is to apply the same transformation to all channels.
```
In IM v7, "all channels" includes alpha, if you have an alpha channel. So your command increases alpha, which makes the outut more opaque. To prevent alpha from changing:
```
magick in.png -channel RGB -brightness-contrast 20x0 +channel out.png
```

**Reply by yellowbooth** [reply]
That is fantastic - thank you so much for your help!

**Reply by yellowbooth** [reply]
I am also trying to stitch the PNGs into a GIF using the below command and finding that the PNG isn't clearing after each frame:
magick podout*.png -channel RGB  -dispose 2  +channel  anim2.gif
In GIMP I have the setting of "Frame disposal where unspecified: One frame per layer (replace)" which works. Is there a similar setting in magick?

**Comment by snibgo** [comment]
I don't know what you mean by "the PNG isn't clearing after each frame".

When a frame contains transparent pixels, what should the output GIF show? Perhaps the colour of the previous frame. Or perhaps something else.

It might help if you upload some sample input PNGs.

EDIT: Perhaps the problem is "-dispose 2". This sets the disposal method for images that haven't already been read into memory. For images that are already in memory, use "-set dispose 2". See the documentation https://imagemagick.org/script/command-line-options.php#dispose

**Comment by yellowbooth** [comment]
If you set it to clear after each frame, it should show what is in the current frame only. It is, however, showing what was in the previous frame, which is stacking up. Here are the first 5 frames
![podout-0](https://user-images.githubusercontent.com/15817969/120936198-8d53bc00-c6fe-11eb-823f-c895079a2e02.png)
![podout-1](https://user-images.githubusercontent.com/15817969/120936199-8e84e900-c6fe-11eb-80fc-1f4d6986a50e.png)
![podout-2](https://user-images.githubusercontent.com/15817969/120936203-8f1d7f80-c6fe-11eb-8a02-2bf946a459d3.png)
![podout-3](https://user-images.githubusercontent.com/15817969/120936204-8f1d7f80-c6fe-11eb-874f-e3722201164f.png)
![podout-4](https://user-images.githubusercontent.com/15817969/120936205-8fb61600-c6fe-11eb-9413-f09ed31f9411.png)

**Comment by snibgo** [comment]
As I suspected, the problem is your "-dispose 2". Either put that _before_ reading the PNG files, or use "-set dispose 2" _after_ reading the PNG files. See the edit in my previous post.

**Comment by yellowbooth** [comment]
Works perfectly - thank you very much :)
